Adopt Cookbook

xorima opened the pull request:
# Description

Standardise everything

Signed-off-by: Jason Field <jason@avon-lea.co.uk>


## Issues Resolved

#31 

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

xorima committed:
Addopt Cookbook

Standardise everything

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
Code owners

xorima committed:
mdl fixes

xorima committed:
Create unit testing structure

kitchen-porter commented:
<!--
  0 Errors
  1 Warning: This is a big Pull Request.
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

xorima committed:
Test all supported OS's

damacus commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,76 @@
+---
+name: ci
+
+on:
```
Is this _still_ broken in that repo?

xorima replied:
ah, this is a manual copy and paste job, we can't automate this due to complexities. 

Why not do it the other way and have yamllint ignore this warning?

damacus replied:
Mostly because it's an easy fix and on does mean something different in yaml. So it's a good habit to get into

damacus commented on metadata.rb:
```diff
@@ -1,13 +1,21 @@
+supports 'scientific'
+supports 'oracle'
 
 depends 'compat_resource', '~> 12.14'
```
```suggestion
```

xorima committed:
Update metadata.rb

Co-authored-by: Dan Webb <dan.webb@damacus.io>

damacus reviewed (changes requested):
As we're bumping the version we can get rid of this.

xorima commented:
And done

xorima committed:
Fix yamllint warning

xorima committed:
Merge branch 'feature/addoption' of github.com:sous-chefs/autofs into feature/addoption

xorima commented:
@damacus could you look at that one failing test on the rhel style boxes, it is `System Package nfs-utils` which works fine on a full fat vm, so it must be a dokken thing, however if I run dokken on my laptop is crashes :(

damacus commented:
I can do tomorrow. Knee deep in `bridge` :)

nvwls commented:
Please allow chef-12.21 to still run

xorima committed:
Migrate to single test cookbook

- Migrate all tests into a single test cookbook
- Change to minumum of chef 12.21

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
Fix kitchen file typo

xorima commented:
Plan to issue another PR after this to move around documentation and examples, but want to get something shipped.

tas50 commented on README.md:
```diff
@@ -1,48 +1,44 @@
-- Network accessible package repositories
+## Requirements
+
+- Chef 15 or higher
```
```suggestion
- Chef Infra Client 12.21 or higher
```

xorima committed:
Update README.md

Co-authored-by: Tim Smith <tsmith@chef.io>

xorima commented:
v2.3.0 shipped
